File: frontend/src/api/chart.ts
/**
 * 图表相关API
 */

import apiClient from './client'

export interface ChartGenerateRequest {
  stock_code: string
  stock_name?: string
  start_date?: string
  end_date?: string
  strategy_name?: string
  strategy_params?: Record<string, any>
  chart_type?: string  // 'kline' or 'kline_with_ma'
  ma_periods?: number[]
}

export interface ChartGenerateResponse {
  chart_id: string
  chart_url: string
  message: string
}

export interface ChartEmbedResponse {
  chart_url: string
  embed_code: string
}

export const chartAPI = {
  /**
   * 生成图表
   */
  generateChart: async (request: ChartGenerateRequest): Promise<ChartGenerateResponse> => {
    const response = await apiClient.post<ChartGenerateResponse>('/api/chart/generate', request)
    return response.data
  },

  /**
   * 获取图表嵌入代码
   */
  getChartEmbedCode: async (chartId: string): Promise<ChartEmbedResponse> => {
    const response = await apiClient.get<ChartEmbedResponse>(`/api/chart/embed/${chartId}`)
    return response.data
  },
}

File: backend/app/models/chart.py
"""图表相关模型"""

from pydantic import BaseModel
from typing import Optional, List, Dict, Any


class ChartGenerateRequest(BaseModel):
    """图表生成请求模型"""
    stock_code: str
    stock_name: Optional[str] = None
    start_date: Optional[str] = None
    end_date: Optional[str] = None
    strategy_name: Optional[str] = None
    strategy_params: Dict[str, Any] = {}
    chart_type: str = "kline"  # 'kline' or 'kline_with_ma'
    ma_periods: Optional[List[int]] = None


class ChartGenerateResponse(BaseModel):
    """图表生成响应模型"""
    chart_id: str
    chart_url: str
    message: str


class ChartEmbedResponse(BaseModel):
    """图表嵌入响应模型"""
    chart_url: str
    embed_code: str
